src/hooks/core/chat/useSystemMessage.ts

```diff
@@ -133,6 +138,19 @@ const useSystemMessage = () => {
         break;
       }
 
+      if (
+        __isMessageOnlyForSystemUser(sortedMessages[0]) &&
+        !__isMySystemMessage(sortedMessages[0])
+      ) {
+        sortedMessages.shift();
+        continue;
+      }
+
+      if (__isMessageForOtherUser(sortedMessages[0])) {
+        firstMessage = checkSystemMessageOnlyForOtherSystemUser(sortedMessages[0]);
+        break;
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This block of code seems to have a lot of conditions and it might be hard to understand what each condition does. It would be better to refactor this into smaller functions with descriptive names that explain what each condition is checking for. This will improve readability and maintainability of the code.

For example, you could create a function `isMessageForSystemUserAndNotMine` that encapsulates the logic in lines 141-143 and another function `isMessageForOtherUser` that encapsulates the logic in line 149.

```diff
-      if (
-        __isMessageOnlyForSystemUser(sortedMessages[0]) &&
-        !__isMySystemMessage(sortedMessages[0])
-      ) {
+      if (isMessageForSystemUserAndNotMine(sortedMessages[0])) {
         sortedMessages.shift();
         continue;
       }
 
-      if (__isMessageForOtherUser(sortedMessages[0])) {
+      if (isMessageForOtherUser(sortedMessages[0])) {
         firstMessage = checkSystemMessageOnlyForOtherSystemUser(sortedMessages[0]);
         break;
       }
```
Please note that these changes assume that the functions `isMessageForSystemUserAndNotMine` and `isMessageForOtherUser` are defined elsewhere in the code. If they are not, you would need to define them.


<!-- This is an auto-generated comment by OSS CodeRabbit -->